pkg/datapath/loader/loader.go

```diff
@@ -99,10 +109,150 @@ func removeEndpointRoute(ep datapath.Endpoint, ip net.IPNet) error {
+	nullStrings := make(map[string]string)
+	for _, v := range strings {
+		nullStrings[v] = v
+	}
+	return nullStrings
+}
+
+// Since we attach the host endpoint datapath to two different interfaces, we
+// need two different NODE_MAC values. patchHostNetdevDatapath creates a new
+// object file for the native device, from the object file for the host device
+// (cilium_host).
+// Since the two object files should only differ by the values of their
+// NODE_MAC symbols, we can avoid a full compilation.
+func patchHostNetdevDatapath(ep datapath.Endpoint, objPath, dstPath, ifName string,
+	nodePortIPv4Addrs, nodePortIPv6Addrs map[string]net.IP) error {
+	hostObj, err := elf.Open(objPath)
+	if err != nil {
+		return err
+	}
+	defer hostObj.Close()
+
+	opts, strings := ELFSubstitutions(ep)
+
+	// The NODE_MAC value is specific to each attachment interface.
+	mac, err := link.GetHardwareAddr(ifName)
+	if err != nil {
+		return err
+	}
+	opts["NODE_MAC_1"] = sliceToBe32(mac[0:4])
+	opts["NODE_MAC_2"] = uint32(sliceToBe16(mac[4:6]))
+
+	ifIndex, err := link.GetIfIndex(ifName)
+	if err != nil {
+		return err
+	}
+
+	if option.Config.EnableNodePort {
+		// First device from the list is used for direct routing between nodes
+		directRoutingIface := option.Config.Devices[0]
+		directRoutingIfIndex, err := link.GetIfIndex(ifName)
+		if err != nil {
+			return err
+		}
+		opts["DIRECT_ROUTING_DEV_IFINDEX"] = directRoutingIfIndex
+		opts["NATIVE_DEV_IFINDEX"] = ifIndex
+		if option.Config.EnableIPv4 {
+			ipv4 := nodePortIPv4Addrs[directRoutingIface]
+			opts["IPV4_DIRECT_ROUTING"] = byteorder.HostSliceToNetwork(ipv4, reflect.Uint32).(uint32)
+			ipv4 = nodePortIPv4Addrs[ifName]
+			opts["IPV4_NODEPORT"] = byteorder.HostSliceToNetwork(ipv4, reflect.Uint32).(uint32)
+		}
+		if option.Config.EnableIPv6 {
+			directRoutingIPv6 := nodePortIPv6Addrs[directRoutingIface]
+			opts["IPV6_DIRECT_ROUTING_1"] = sliceToBe32(directRoutingIPv6[0:4])
+			opts["IPV6_DIRECT_ROUTING_2"] = sliceToBe32(directRoutingIPv6[4:8])
+			opts["IPV6_DIRECT_ROUTING_3"] = sliceToBe32(directRoutingIPv6[8:12])
+			opts["IPV6_DIRECT_ROUTING_4"] = sliceToBe32(directRoutingIPv6[12:16])
+			nodePortIPv6 := nodePortIPv6Addrs[ifName]
+			opts["IPV6_NODEPORT_1"] = sliceToBe32(nodePortIPv6[0:4])
+			opts["IPV6_NODEPORT_2"] = sliceToBe32(nodePortIPv6[4:8])
+			opts["IPV6_NODEPORT_3"] = sliceToBe32(nodePortIPv6[8:12])
+			opts["IPV6_NODEPORT_4"] = sliceToBe32(nodePortIPv6[12:16])
+		}
+	}
+
+	// Among string substitutions, only the calls map name is specific to each
+	// attachment interface.
+	strings = nullifyStringSubstitutions(strings)
+	callsMapHostDevice := bpf.LocalMapName(callsmap.HostMapName, uint16(ep.GetID()))
+	strings[callsMapHostDevice] = bpf.LocalMapName(callsmap.NetdevMapName, uint16(ifIndex))
+
+	return hostObj.Write(dstPath, opts, strings)
+}
+
+func (l *Loader) reloadHostDatapath(ctx context.Context, ep datapath.Endpoint, objPath string) error {
```

Comment:
It is not THE template for the host endpoint. It's the object file that will be attached to the host device (i.e., `cilium_host`). That object file was derived from the host endpoint located in `state/template/` as usual. That being said, the principle is the same since we derive patched object files for native devices from the object file for the host device. Not sure if that's enough to address your comment, but I will add a function comment to make this clearer.